arcgis-ios-sdk-samples/Display information/Sketch on the map/SketchViewController.swift

```diff
@@ -16,76 +16,82 @@ import UIKit
-        // Enable/disable UI elements appropriately
-        self.undoBBI.isEnabled = self.sketchEditor.undoManager.canUndo
-        self.redoBBI.isEnabled = self.sketchEditor.undoManager.canRedo
-        self.clearBBI.isEnabled = self.sketchEditor.geometry != nil && !self.sketchEditor.geometry!.isEmpty
     }
     
     // MARK: - Actions
     
-    @IBAction func geometryValueChanged(_ segmentedControl: UISegmentedControl) {
-        switch segmentedControl.selectedSegmentIndex {
-        case 0:// point
-            self.sketchEditor.start(with: nil, creationMode: .point)
-            
-        case 1:// polyline
-            self.sketchEditor.start(with: nil, creationMode: .polyline)
-            
-        case 2:// freehand polyline
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolyline)
-            
-        case 3:// polygon
-            self.sketchEditor.start(with: nil, creationMode: .polygon)
-            
-        case 4:// freehand polygon
-            self.sketchEditor.start(with: nil, creationMode: .freehandPolygon)
-            
-        default:
-            break
+    @IBAction func addGeometry() {
+        // Create an alert controller for the action sheets.
+        let alertController = UIAlertController(title: "Select a creation mode", message: nil, preferredStyle: .actionSheet)
+        // Key value pairs containing the creation modes and their titles.
+        let creationModes: KeyValuePairs = [
+            "Arrow": AGSSketchCreationMode.arrow,
+            "Ellipse": .ellipse,
+            "FreehandPolygon": .freehandPolygon,
+            "FreehandPolyline": .freehandPolyline,
+            "Multipoint": .multipoint,
+            "Point": .point,
+            "Polygon": .polygon,
+            "Polyline": .polyline,
+            "Rectangle": .rectangle,
+            "Triangle": .triangle
+        ]
+        // Create an action for each creation mode and add it to the alert controller.
+        creationModes.forEach { creationMode in
+            let action = UIAlertAction(title: creationMode.key, style: .default) { (_) in
+                self.sketchEditor.start(with: nil, creationMode: creationMode.value)
+            }
+            alertController.addAction(action)
         }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
         
+        // Present the action sheets when the add button is tapped.
+        alertController.popoverPresentationController?.barButtonItem = addBarButtonItem
+        present(alertController, animated: true)
+        // Reset the sketch editor.
         self.mapView.sketchEditor = self.sketchEditor
     }
     
     @IBAction func undo() {
-        if self.sketchEditor.undoManager.canUndo { // extra check, just to be sure
+        // Check if there are actions to undo.
+        if self.sketchEditor.undoManager.canUndo {
```

Comment:
I'll leave this open for discussion. I don't see any harm in leaving it as is

Reply:
Calling `self.sketchEditor.undoManager.undo()` without anything on the "undo" stack doesn't cause any issues (Xcode 13, iOS 15 simulator).  The `NSUndoManager` doc doesn't say anything about any issues, but I'd leave the check in, just in case...